src/server/campaignd/server.cpp

```diff
@@ -1891,14 +1916,28 @@ void server::handle_change_passphrase(const server::request& req)
 	}
 }
 
-bool server::authenticate_forum(const config& addon, const std::string& passphrase) {
+bool server::authenticate_forum(const config& addon, const std::string& passphrase, bool is_delete) {
 	if(!user_handler_) {
 		return false;
 	}
 
-	std::string author = addon["author"].str();
+	std::string uploader = addon["uploader"].str();
+	std::string id = addon["name"].str();
+	bool do_authors_exist = user_handler_->db_do_any_authors_exist(server_id_, id);
+	bool is_primary = user_handler_->db_is_user_primary_author(server_id_, id, uploader);
+	bool is_secondary = user_handler_->db_is_user_secondary_author(server_id_, id, uploader);
+
+	// allow if there is no author information - this is a new upload
+	// don't allow other people to upload if author information does exist
+	// don't allow secondary authors to remove the add-on from the server
+	if((do_authors_exist && !is_primary && !is_secondary) || (is_secondary && is_delete)) {
```

Comment:
Is it guaranteed that if there is no author info that this must be a new upload?

Reply:
Technically no, since it would also fail to find author info for existing add-ons using `forum_auth=yes` prior to this PR being merged. However, the only add-ons currently using that are test add-ons I uploaded previously, so it would not affect any real add-ons created by anyone else.

Reply:
I was rather wondering about uploads that did not use forum_auth so far but I see now that that is rejected earlier.